src/BloomExe/Publish/PdfViewer.cs

```diff
@@ -99,9 +99,17 @@ public bool ShowPdf(string pdfFile)
 				// and continue to show it using that.
 			}
 #endif
-
-			var url = string.Format("{0}{1}?file=/bloom/{2}", Bloom.web.ServerBase.PathEndingInSlash,
-				FileLocator.GetFileDistributedWithApplication("pdf/web/viewer.html"), pdfFile);
+			// Escaping the filename twice for characters like # is needed in order to get the
+			// pdf filename through to our local server on linux.  This is to prevent the filename
+			// from being cut short at the # character.  See https://jira.sil.org/browse/BL-951.
+			// (You may regard this as a hack to get around a bug in the Linux xulrunner.)
```

Comment:
BL-951 doesn't really explain why we need to do it twice, like you did in a response to an earlier comment. Especially since other comments make a point that we need to call Unescape exactly as many times as Escape, I think it would be worth adding the information about how the two levels of escaping get appropriately undone.